foundation/org.eclipse.persistence.core/src/main/java/org/eclipse/persistence/internal/queries/ExpressionQueryMechanism.java

```diff
@@ -835,7 +835,16 @@ protected SQLUpdateStatement buildUpdateStatement(DatabaseTable table) {
         updateStatement.setModifyRow(getModifyRow());
         updateStatement.setTranslationRow(getTranslationRow());
         if (getDescriptor().hasReturningPolicies() && getDescriptor().getReturnFieldsToGenerateUpdate() != null) {
-            updateStatement.setReturnFields(getDescriptor().getReturnFieldsToGenerateUpdate());
+            // In case of RelationalDescriptor only return fields for current table must be used.
+            Vector<DatabaseField> returnFieldsForTable = new NonSynchronizedVector();
```

Comment:
Would it be possible to start using common `List` (`ArrayList`) where possible instead? I can see no reason for deprecated `Vector` here...

Reply:
OK replaced by `List` (`ArrayList`)